Jenkinsfile

```diff
@@ -16,87 +16,316 @@
+  sh """
+    for valuesFile in ${HELM_VALUES_DIR}/*.yaml; do
+      USAGE=${USAGE} envsubst < \$valuesFile > \$valuesFile~gen
+    done
+  """
+}
+
+def rolloutStatus(kind, name, timeout, namespace) {
+  sh """
+    kubectl rollout status ${kind} ${name} \
+      --timeout=${timeout} \
+      --namespace=${namespace}
+  """
+}
+
+def rolloutStatusMongoDB() {
+  rolloutStatus('deployment', "${MONGODB_CHART_NAME}", "${ROLLOUT_STATUS_TIMEOUT}", "${NAMESPACE}")
+}
+
+def rolloutStatusElasticsearch() {
+  rolloutStatus('statefulset', "${ELASTICSEARCH_CHART_NAME}-master", "${ROLLOUT_STATUS_TIMEOUT}", "${NAMESPACE}")
+}
+
+def rolloutStatusKafka() {
+  rolloutStatus('statefulset', "${KAFKA_CHART_NAME}", "${ROLLOUT_STATUS_TIMEOUT}", "${NAMESPACE}")
+}
+
+def rolloutStatusNuxeo() {
+  rolloutStatus('deployment', "${NUXEO_CHART_NAME}", "${ROLLOUT_STATUS_TIMEOUT}", "${NAMESPACE}")
+}
+
+def isTriggered() {
+  return !currentBuild.upstreamBuilds.isEmpty()
+}
+
+def isTriggeredByNuxeoPR() {
+  return isTriggered() && params.NUXEO_VERSION?.trim()
+}
+
+def upstreamJobName = isTriggered() ? currentBuild.upstreamBuilds[0].getFullProjectName() : null
+def upstreamBuildNumber = isTriggered() ? currentBuild.upstreamBuilds[0].getNumber() : null
+
 pipeline {
+
   agent {
     label "jenkins-nodejs"
   }
+
+  parameters {
+    string(name: 'NUXEO_VERSION', defaultValue: '', description: 'Version of the Nuxeo server image, defaults to 11.x.')
+    string(name: 'NUXEO_REPOSITORY', defaultValue: '', description: 'GitHub repository of the nuxeo project.')
+    string(name: 'NUXEO_SHA', defaultValue: '', description: 'Git commit sha of the nuxeo/nuxeo upstream build.')
+  }
+
   environment {
-    HELM_CHART_REPOSITORY_NAME = 'local-jenkins-x'
-    HELM_CHART_REPOSITORY_URL = 'http://jenkins-x-chartmuseum:8080'
-    HELM_CHART_NUXEO = 'nuxeo'
-    HELM_RELEASE_NUXEO = 'rest-api-tests'
-    NAMESPACE_NUXEO = "nuxeo-platform-rest-api-tests-${BUILD_NUMBER}"
-    SERVICE_NUXEO = "${HELM_RELEASE_NUXEO}-${HELM_CHART_NUXEO}"
-    SERVICE_DOMAIN = ".${NAMESPACE_NUXEO}.svc.cluster.local"
-    SERVICE_ACCOUNT = 'jenkins'
+    BITNAMI_CHART_REPOSITORY_NAME = 'bitnami'
+    BITNAMI_CHART_REPOSITORY_URL = 'https://charts.bitnami.com/bitnami'
+    ELASTIC_CHART_REPOSITORY_NAME = 'elastic'
+    ELASTIC_CHART_REPOSITORY_URL = 'https://helm.elastic.co/'
+    NUXEO_CHART_REPOSITORY_NAME = 'nuxeo'
+    NUXEO_CHART_REPOSITORY_URL = 'https://chartmuseum.platform.dev.nuxeo.com/'
+    MONGODB_CHART_NAME = 'mongodb'
+    MONGODB_CHART_VERSION = '7.14.2'
+    ELASTICSEARCH_CHART_NAME = 'elasticsearch'
+    ELASTICSEARCH_CHART_VERSION = '7.9.2'
+    KAFKA_CHART_NAME = 'kafka'
+    KAFKA_CHART_VERSION = '11.8.8'
+    NUXEO_CHART_NAME = 'nuxeo'
+    NUXEO_CHART_VERSION = '~2.0.0'
+    NUXEO_DOCKER_REPOSITORY = "${isTriggeredByNuxeoPR() ? DOCKER_REGISTRY : PUBLIC_DOCKER_REGISTRY}/nuxeo/nuxeo"
+    NUXEO_VERSION = "${params.NUXEO_VERSION?.trim() ? params.NUXEO_VERSION : '11.x'}"
```

Comment:
Why not using the parameter default value?

Reply:
To detect if the trigger was not form a PR with `isTriggeredByNuxeoPR`, in which case the param should be empty.